src/platform/nrfconnect/Reboot.cpp

```diff
@@ -28,19 +28,7 @@
 namespace chip {
 namespace DeviceLayer {
 
-#if defined(CONFIG_ARCH_POSIX) || defined(CONFIG_SOC_SERIES_NRF54HX)
```

Comment:
Why is this change needed?

Reply:
Because in nrf54h20 case we don't use Reboot function right? we use dfu_target_suit_reboot. Reset reason is saved in RESETINFO register, so we also don't need to use GetSoftwareRebootReason(), so I thought it is unnecessary to define these functions in case of using nrf54h20.